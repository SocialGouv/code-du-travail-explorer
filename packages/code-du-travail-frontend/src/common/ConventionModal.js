import React from "react";
import dynamic from "next/dynamic";
import { Button } from "@cdt/ui";
import { DialogContent, DialogOverlay } from "@reach/dialog";

// prevent ConventionPicker in initial bundle
const _ConventionPicker = dynamic(() => import("./ConventionPicker"), {
  loading: () => <p>Chargement des conventions...</p>
});

class ConventionModal extends React.Component {
  state = {
    modalIsOpen: false
  };

  inputRef = React.createRef();

  openModal = e => {
    e.preventDefault();
    this.setState({
      modalIsOpen: true
    });
  };

  closeModal = () => {
    this.setState({
      modalIsOpen: false
    });
  };

  render() {
    const text = this.props.text || "Trouvez votre convention collective";
    return (
      <React.Fragment>
        <Button link onClick={this.openModal}>
          {text}
        </Button>
        <DialogOverlay
          isOpen={this.state.modalIsOpen}
          style={{ background: "rgba(0, 0, 0, .5)" }}
          initialFocusRef={this.inputRef}
          onDismiss={this.closeModal}
        >
          <DialogContent style={{ borderRadius: "3px", margin: "5vh auto" }}>
            <h2>Convention collective</h2>
            <div>
              Saisissez l&apos;identifiant de convention collective (IDCC), le
              nom de la branche, ou le code NAF :
              <div style={{ marginTop: 10 }}>
                <_ConventionPicker ref={this.inputRef} />
              </div>
              <div>
                <div style={{ marginTop: 20, fontWeight: "bold" }}>
                  Comment trouver ma convention collective ?
                </div>
                <li>
                  Elle doit figurer au <b>bulletin de paie</b>, et sur une
                  notice remise à l&apos;embauche ou{" "}
                  <b>sur le contrat de travail</b>
                </li>
                <li>
                  Sur l&apos;affichage obligatoire dans l&apos;entreprise (avec
                  les modalités de consultation en entreprise)
                </li>
                <div style={{ marginTop: 20, fontWeight: "bold" }}>
                  Quelle est la CCN applicable ?
                </div>
                <li>
                  En principe celle relevant de l&apos;activité principale dans
                  l&apos;entreprise pour les cas les plus simples.
                </li>
              </div>
            </div>
          </DialogContent>
        </DialogOverlay>
      </React.Fragment>
    );
  }
}

export default ConventionModal;
