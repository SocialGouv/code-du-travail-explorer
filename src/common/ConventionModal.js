import React from "react";
import Modal from "react-modal";
import dynamic from "next/dynamic";

// prevent ConventionPicker in initial bundle
const _ConventionPicker = dynamic(() => import("./ConventionPicker"), {
  loading: () => <p>Chargement des conventions...</p>
});

if (typeof document !== "undefined") {
  Modal.setAppElement("body");
}

const modalStyles = {
  overlay: {
    zIndex: 10000,
    backgroundColor: "rgba(0,0,0,0.8)"
  },
  content: {
    top: "50%",
    left: "50%",
    right: "auto",
    bottom: "auto",
    overflow: "visible",
    marginRight: "-50%",
    transform: "translate(-50%, -50%)",
    width: "70%",
    maxHeight: "90%"
  }
};

class ConventionModal extends React.Component {
  state = {
    modalIsOpen: false
  };

  openModal = e => {
    e.preventDefault();
    this.setState({
      modalIsOpen: true
    });
  };

  closeModal = () => {
    this.setState({
      modalIsOpen: false
    });
  };

  render() {
    const text = this.props.text || "Trouvez votre convention collective";
    return (
      <React.Fragment>
        <a
          href="#"
          onClick={this.openModal}
          dangerouslySetInnerHTML={{ __html: text }}
        />
        <Modal
          style={modalStyles}
          isOpen={this.state.modalIsOpen}
          onRequestClose={this.closeModal}
        >
          <h2>Convention collective</h2>
          <div>
            Trouvez votre convention collective :
            <div style={{ marginTop: 10 }}>
              <_ConventionPicker />
            </div>
          </div>
        </Modal>
      </React.Fragment>
    );
  }
}

export default ConventionModal;
