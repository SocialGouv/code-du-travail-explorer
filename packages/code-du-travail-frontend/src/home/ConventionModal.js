import React from "react";
import { DialogContent, DialogOverlay } from "@reach/dialog";
import styled from "styled-components";

import { Outil, OutilCard } from "./Outils";
import { ConventionForm } from "../common/ConventionForm";
import { CompanyForm } from "../common/CompanyForm";
import {
  searchIdcc,
  searchCompanies,
  getCompany
} from "../common/convention.service";

class ConventionModal extends React.Component {
  state = {
    modalIsOpen: false
  };
  openModal = e => {
    e.preventDefault();
    this.setState({
      modalIsOpen: true
    });
  };

  closeModal = () => {
    this.setState({
      modalIsOpen: false
    });
  };

  render() {
    return (
      <OutilCard>
        <Button onClick={this.openModal}>
          <Outil
            title="Votre convention collective"
            text="Trouvez simplement la convention collective dont vous dÃ©pendez"
            icon="/static/assets/icons/book_web.svg"
          />
        </Button>
        <DialogOverlay
          isOpen={this.state.modalIsOpen}
          style={{ background: "rgba(0, 0, 0, .5)" }}
          onDismiss={this.closeModal}
        >
          <DialogContent style={{ borderRadius: "3px", margin: "5vh auto" }}>
            <ConventionForm onSearch={searchIdcc} />
            <CompanyForm onSearch={searchCompanies} getCompany={getCompany} />
          </DialogContent>
        </DialogOverlay>
      </OutilCard>
    );
  }
}

export default ConventionModal;

const Button = styled.button`
  background: transparent;
  border: none;
  display: flex;
  flex-direction: column;
  align-items: center;
  font-family: inherit;
  color: inherit;
`;
