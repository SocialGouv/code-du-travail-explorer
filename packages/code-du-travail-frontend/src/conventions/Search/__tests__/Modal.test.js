import React from "react";
import { render } from "@testing-library/react";
import ConventionModal from "../Modal";

// Trouvez votre convention collective

describe("<ConventionModal />", () => {
  it("should render", () => {
    const { container } = render(<ConventionModal />);
    expect(container).toMatchSnapshot();
  });

  it("should render a popup when click on button", () => {
    const { baseElement, getByText, getByPlaceholderText } = render(
      <ConventionModal />
    );
    const button = getByText(/votre convention collective/i);
    button.click();
    const input = getByPlaceholderText(
      /Nom d'entreprise, SIRET, nom de convention collective/i
    );
    expect(input).toBeTruthy();
    expect(baseElement).toMatchSnapshot();
  });

  it("should close the modal", async () => {
    const { getByText, queryByPlaceholderText } = render(<ConventionModal />);
    const button = getByText(/votre convention collective/i);
    button.click();
    const el = document.body.querySelector("[data-reach-dialog-overlay]");
    el.click();
    expect(
      queryByPlaceholderText(
        /Nom d'entreprise, SIRET, nom de convention collective/i
      )
    ).toBeNull();
  });
});
