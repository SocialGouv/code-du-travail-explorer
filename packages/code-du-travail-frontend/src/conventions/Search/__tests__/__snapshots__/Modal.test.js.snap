// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConventionModal /> should render 1`] = `
.c1 {
  padding: 0.625rem 1rem;
  border: 1px solid #c9d3df;
  border-radius: 0.25rem;
  background: #fff;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 100%;
}

.c4 {
  color: #4c5467;
  font-size: 0.875rem;
}

.c3 {
  color: #005994;
  font-size: 1.125rem;
}

.c2 {
  width: 2.5rem;
  display: inline-block;
  margin: 1rem;
}

.c0 {
  padding: 0;
  width: 100%;
  height: 100%;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: transparent;
  border: none;
  border-radius: 0.25rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  cursor: pointer;
}

.c0 > * {
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
}

.c0:focus > *,
.c0:active > *,
.c0:hover > * {
  border: 1px solid #80bdff;
  box-shadow: 0 0 2px 2px #e5f1fe;
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

<div>
  <button
    class="c0"
  >
    <div
      class="c1"
    >
      <img
        alt=""
        class="c2"
        src="/static/assets/icons/book_web.svg"
      />
      <h3
        class="c3"
      >
        Votre convention collective
      </h3>
      <p
        class="c4"
      >
        Trouvez simplement la convention collective dont vous dépendez
      </p>
    </div>
  </button>
</div>
`;

exports[`<ConventionModal /> should render a popup when click on button 1`] = `
.c10:not(:focus):not(:active) {
  position: absolute;
  width: 1px;
  height: 1px;
  margin: -1px;
  padding: 0;
  overflow: hidden;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  border: 0;
}

.c5 {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background: rgba(0,0,0,0.5);
  overflow: auto;
}

.c6 {
  position: relative;
  margin: 10vh auto;
  padding: 2rem;
  width: 50vw;
  max-height: calc(100vh - 2 * 10vh);
  background: white;
  border-radius: 0.25rem;
  outline: none;
  overflow-y: auto;
}

.c9 {
  position: absolute;
  top: 0;
  right: 0;
  padding: 0.625rem;
  color: #8393a7;
  line-height: 0;
  background: transparent;
  border: none;
  -webkit-appearance: none;
  cursor: pointer;
}

.c7 {
  max-width: 1200px;
  margin: 0 auto;
  padding: 0 1.25rem;
}

.c7 > *:first-child {
  margin-top: 0;
}

.c7 > *:last-child {
  margin-bottom: 0;
}

.c1 {
  padding: 0.625rem 1rem;
  border: 1px solid #c9d3df;
  border-radius: 0.25rem;
  background: #fff;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 100%;
}

.c4 {
  color: #4c5467;
  font-size: 0.875rem;
}

.c3 {
  color: #005994;
  font-size: 1.125rem;
}

.c2 {
  width: 2.5rem;
  display: inline-block;
  margin: 1rem;
}

.c8 {
  width: 100%;
}

.c0 {
  padding: 0;
  width: 100%;
  height: 100%;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: transparent;
  border: none;
  border-radius: 0.25rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  cursor: pointer;
}

.c0 > * {
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
}

.c0:focus > *,
.c0:active > *,
.c0:hover > * {
  border: 1px solid #80bdff;
  box-shadow: 0 0 2px 2px #e5f1fe;
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

@media (max-width:1200px) {
  .c6 {
    width: 60vw;
  }
}

@media (max-width:980px) {
  .c6 {
    width: 80vw;
  }
}

@media (max-width:600px) {
  .c6 {
    margin: 1rem;
    max-height: calc(100vh - 2 * 1rem);
    width: calc(100% - 2 * 1rem);
  }
}

@media (max-width:600px) {
  .c9 {
    position: fixed;
    top: 0.625rem;
    right: 0.625rem;
  }
}

@media print {
  .c7 {
    max-width: 100%;
    padding: 0;
  }
}

<body>
  <div
    aria-hidden="true"
  >
    <button
      class="c0"
    >
      <div
        class="c1"
      >
        <img
          alt=""
          class="c2"
          src="/static/assets/icons/book_web.svg"
        />
        <h3
          class="c3"
        >
          Votre convention collective
        </h3>
        <p
          class="c4"
        >
          Trouvez simplement la convention collective dont vous dépendez
        </p>
      </div>
    </button>
  </div>
  <reach-portal>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="1"
    />
    <div
      data-focus-lock-disabled="false"
    >
      <div>
        <div
          class="c5"
          data-reach-dialog-overlay="true"
        >
          <div
            aria-modal="true"
            class="c6"
            data-reach-dialog-content="true"
            tabindex="-1"
          >
            <div
              class="c7"
            >
              <h3>
                Recherche de convention collective
              </h3>
              <p>
                Afin de sélectionner votre convention collective, saisissez le nom de votre entreprise, son numéro SIRET, ou directement le nom de votre convention collective.
              </p>
              <input
                class="c8"
                placeholder="Nom d'entreprise, SIRET, nom de convention collective"
                type="search"
                value=""
              />
              
            </div>
            <button
              class="c9"
              title="fermer la modale"
            >
              <div
                class="c10"
              >
                fermer la modale
              </div>
              <svg
                aria-hidden="true"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <line
                  x1="18"
                  x2="6"
                  y1="6"
                  y2="18"
                />
                <line
                  x1="6"
                  x2="18"
                  y1="6"
                  y2="18"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </reach-portal>
</body>
`;
